plugins/slo/pkg/slo/validate.go

```diff
@@ -16,18 +21,33 @@ func matchEnum(target string, enum []string, returnErr error) error {
 
 /// Validates Input based on the necessities of our preconfigured formant,
 /// NOT validating the OpenSLO / Sloth format
-func ValidateInput(slo *api.ServiceLevelObjective) error {
-	if slo.GetId() == "" {
-		return shared.ErrInvalidId
+func ValidateInput(slorequest *api.CreateSLORequest) error {
+	if len(slorequest.Services) == 0 {
+		return shared.ErrMissingServices
 	}
-	if err := validateSLODescription(slo.GetDescription()); err != nil {
+	if err := validateSLODatasource(slorequest.SLO.GetDatasource()); err != nil {
 		return err
 	}
-	if err := validateSLODatasource(slo.GetDatasource()); err != nil {
+	if err := validateAlert(slorequest.SLO.GetAlerts()); err != nil {
 		return err
 	}
-	if err := validateAlert(slo.GetAlerts()); err != nil {
-		return err
+	// Expected monitor window
+	validWindows := []string{"7d", "28d", "30d"}
+	if !slices.Contains(validWindows, slorequest.SLO.GetMonitorWindow()) {
+		return shared.ErrInvalidMonitorWindow
+	}
+
+	budgetingIntervalRegex := regexp.MustCompile(`(^\d+)[m]$`)
```

Comment:
If you edit the proto field type (see comment in slo.proto) you won't need this parsing logic anymore. you can also allow arbitrary durations for monitoring windows, and/or compare them against strongly typed duration values instead of strings.

Reply:
Converted budgeting interval to google's protobuf.Duration